src/client/components/pages/RoomAdmin/RoomAdmin.tsx

```diff
@@ -181,32 +251,105 @@ const RoomAdmin: FunctionComponent = (): ReactElement => {
                     <TableHeadingCell scope="col">Capacity</TableHeadingCell>
                     <TableHeadingCell scope="col">Edit</TableHeadingCell>
                   </TableRow>
+                  <TableRow isStriped>
+                    <TableHeadingCell scope="col">
+                      <Dropdown
+                        hideError
+                        id="campus-filter"
+                        label="Change to filter the list of rooms by campus"
+                        isLabelVisible={false}
+                        onChange={
+                          (evt: ChangeEvent<HTMLSelectElement>): void => {
+                            setCampusFilter(evt.target.value as CAMPUS);
+                          }
+                        }
+                        name="campus-filter"
+                        value={campusFilter}
+                        options={
+                          Object.values(CAMPUS)
+                            .map((value) => ({ value, label: value }))
+                        }
+                      />
+
+                    </TableHeadingCell>
+                    <TableHeadingCell scope="col">
+                      <Dropdown
+                        hideError
+                        id="building-filter"
+                        label="Change to filter the list of rooms by building"
+                        isLabelVisible={false}
+                        options={
+                          [{ value: 'All', label: 'All' }, ...buildingDropDown]
+                        }
+                        name="building-filter"
+                        onChange={
+                          (event:React.ChangeEvent<HTMLInputElement>) => {
+                            setBuildingFilter(event.currentTarget.value);
+                          }
+                        }
+                      />
+                    </TableHeadingCell>
+                    <TableHeadingCell scope="col">
+                      <TextInput
+                        id="room-filter"
+                        name="room-filter"
+                        placeholder="Filter by Room Name"
+                        value={roomFilter}
+                        label="Change to filter the list of rooms by selected room"
+                        isLabelVisible={false}
+                        hideError
+                        onChange={
+                          (event:React.ChangeEvent<HTMLInputElement>) => {
+                            setRoomFilter(event.currentTarget.value);
+                          }
+                        }
+                      />
+                    </TableHeadingCell>
+                    <TableHeadingCell scope="col">
+                      <TextInput
+                        id="capacity-filter"
+                        name="capacity-filter"
+                        placeholder="Enter Capacity"
```

Comment:
For consistency with the other filter fields, I think the placeholder should be "Filter by Capacity".